File: apple/Sources/ReactNativeWatchOS/Modifiers/GridCellUnsizedAxes.swift
import SwiftUI

/// SwiftUI `.gridCellUnsizedAxes(_:)`. Opts a cell out of the `Grid`'s
/// width/height sizing along the given axes. Available on watchOS 9+.
enum RNWGridCellUnsizedAxesModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("gridCellUnsizedAxes") { view, params, _ in
            return AnyView(view.gridCellUnsizedAxes(axisSet(params.string("axes"))))
        }
    }

    private static func axisSet(_ s: String?) -> Axis.Set {
        switch s {
        case "horizontal": return .horizontal
        case "vertical":   return .vertical
        case "both":       return [.horizontal, .vertical]
        default:           return []
        }
    }
}

File: src/modifiers/gridCellUnsizedAxes.ts
import { createModifier } from './createModifier';

/// Axes along which the cell should NOT inflate to the row/column size.
export type GridCellUnsizedAxesValue = 'horizontal' | 'vertical' | 'both';

export interface GridCellUnsizedAxesParams {
  /// The axes to leave unsized. Use `'both'` to opt out of grid sizing on
  /// both axes (e.g. for a `Divider` that should keep its intrinsic size).
  axes: GridCellUnsizedAxesValue;
}

/// SwiftUI `.gridCellUnsizedAxes(_:)`. Prevents a cell from filling the
/// width/height the `Grid` would otherwise impose along the given axes.
export function gridCellUnsizedAxes(
  params: GridCellUnsizedAxesParams
): ReturnType<typeof createModifier>;
export function gridCellUnsizedAxes(
  axes: GridCellUnsizedAxesValue
): ReturnType<typeof createModifier>;
export function gridCellUnsizedAxes(
  a: GridCellUnsizedAxesParams | GridCellUnsizedAxesValue
) {
  if (typeof a === 'string') {
    return createModifier('gridCellUnsizedAxes', { axes: a });
  }
  return createModifier('gridCellUnsizedAxes', a);
}
